solr/webapp/web/solrUi/src/app/modules/common/shared.module.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {NgModule} from '@angular/core';
+import {SizeUnitConverterPipe} from "./pipes/sizeconverter/size-unit-converter-pipe.pipe";
+import {TimeAgoPipe} from "./pipes/timeconverter/time-ago.pipe";
+import {CommonModule, NgClass, NgFor, NgIf} from "@angular/common";
+import {IResourceParams} from "@ngx-resource/core";
+import {environment} from "../../../environments/environment";
+
+
+@NgModule({
+  declarations: [SizeUnitConverterPipe, TimeAgoPipe],
+  imports: [CommonModule, NgClass, NgFor, NgIf],
+  exports: [SizeUnitConverterPipe, TimeAgoPipe, NgClass, NgFor, NgIf],
+})
+export class SharedModule {
```

Comment:
This is part of modularity that Eric has talked. That other extensions can be applied on the fly, this is just one of the first step to provide common logic across multiple components and modules within Angular, as well in the future there will Interface/Adapter implemented to support external modules integration to the application.

Reply:
I see. I saw only the static functions in the module and thought "if all its content is static, what is the role of this class to be a module?". I didn't have a context where this module could be used.

It is nice that you thought about extendability of the frontend, I was asked about this too. There are definitely multiple ways to achieve this, but I believe integrating it too early may overcomplicate things.

Reply:
@malliaridis, well If we begin our implementation and only at the later stage add the structure to integrate third party extension, it might require significant amount of work at the latest stage of the development. 
It's simple for example, if we want to integrate multi-laguage support into our UI, it would be easier to add it at this stage of the application development rather at the latest stage.
Regarding this Shared module, is more about code grouping within the application structure.
At this point of the implementation, we have only one component is provided for review.